src/integrationTest/java/com/mongodb/hibernate/BasicCrudTests.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.LIST;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Sorts;
+import com.mongodb.hibernate.internal.cfg.MongoConfigurationBuilder;
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+import org.bson.BsonDocument;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SessionFactoryScopeAware;
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(annotatedClasses = {BasicCrudTests.Book.class, BasicCrudTests.BookWithEmbeddedField.class})
+class BasicCrudTests implements SessionFactoryScopeAware {
+
+    @AutoClose
+    private static MongoClient mongoClient;
+
+    private SessionFactoryScope sessionFactoryScope;
+    private MongoCollection<BsonDocument> collection;
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        var config = new MongoConfigurationBuilder(
+                        sessionFactoryScope.getSessionFactory().getProperties())
+                .build();
+        if (mongoClient == null) {
+            mongoClient = MongoClients.create(config.mongoClientSettings());
+        }
+        collection = mongoClient.getDatabase(config.databaseName()).getCollection("books", BsonDocument.class);
+        collection.drop();
+    }
```

Comment:
[I am leaving this comment just to make sure we don't forget to make the change]

@NathanQingyangXu, as you proposed, we should make `mongoClient` an instance field, and initialize it in an instance `@BeforeAll` method. The latter also applies to the `collection` field. The only thing we need to do in the `@BeforeEach` method is to call `collection.drop();`